code/modules/power/supermatter/supermatter_hypermatter_state.dm

```diff
@@ -0,0 +1,195 @@
+///Handles the main process of the hypermatter
+/obj/machinery/power/supermatter_crystal/proc/handle_hypermatter_state()
+
+	if(!anchored) //Don't unanchor the shards
+		damage += 150
+		hypermatter_state = FALSE
+		update_appearance()
+		return
+
+
+	var/static/list/angles = list(0, 45, 90, 135, 180, 225, 270, 315, 360)
+
+	if(power)
+		power = max(power - 30, 0)
+
+		for(var/_ in 1 to rand(2, 6))
+			if(hypermatter_power_amount < 1000)
+				break
+			var/angle_to_shoot = pick(angles)
+			fire_nuclear_particle(angle_to_shoot, 1.2, 1000, "sm_nuclear_particle")
+			hypermatter_power_amount = max(hypermatter_power_amount - 100, 0)
+	else
+		hypermatter_power_amount = max(hypermatter_power_amount - 200, 0)
+
+
+	if(prob(5))
+		fire_nuclear_particle()
+
+		supermatter_zap(
+				zapstart = src,
+				range = 3,
+				zap_str = 5 * power,
+				zap_flags = ZAP_SUPERMATTER_FLAGS,
+				zap_cutoff = 300,
+				power_level = power,
+			)
+
+	if(hypermatter_power_amount < 1000)
+		hypermatter_cooldown -= 5 SECONDS
+
+/obj/machinery/power/energy_accumulator/nuclear_accumulator
+	name = "nuclear accumulator"
+	desc = "A large, powerful nuclear accumulator. It is capable of storing a large amount of power."
+	icon = 'icons/obj/supermatter.dmi'
+	icon_state = "nuclear_accumulator"
+	circuit = /obj/item/circuitboard/machine/nuclear_accumulator
+	set_dir_on_move = FALSE
+
+/obj/machinery/power/energy_accumulator/nuclear_accumulator/bullet_act(obj/projectile/projectile)
+	if(!istype(projectile, /obj/projectile/energy/nuclear_particle) || projectile.dir != turn(dir, 180))
```

Comment:
why the turn 180?

Reply:
to have accumulator only collect those particles coming from the front